### PR TITLE
[Feature] Visited by

### PR DESCRIPTION
**Pull request by gohabereg**

### CODE REVIEW

File: src/api/events/index.ts [hunk 1]
```diff
@@ -51,3 +52,13 @@ export async function getLatestRepetitions(
...
   return (await api.call(QUERY_LATEST_REPETITIONS, { projectId, eventId })).project.event.repetitions.shift() || null;
 }
+
+/**
+ * Mark event as visited for current user
+ *
+ * @param {string} eventId — visited event
+ * @return {Promise<boolean>}
+ */
+export async function visitEvent(projectId: string, eventId: string): Promise<boolean> {
```
Comment (neSpecc) [inline src/api/events/index.ts]: `projectId` is not described

File: src/components/utils/Badge.vue [hunk 2]
```diff
@@ -72,6 +72,10 @@ export default {
       background-color: var(--color-indicator-critical);
     }
 
+    &--visited {
+      background-color: var(--color-bg-main);
```
Comment (neSpecc) [inline src/components/utils/Badge.vue]: and color second

File: src/components/events/Overview.vue [hunk 3]
```diff
@@ -167,6 +167,14 @@ export default {
...
     });
     this.loading = false;
+  },
+  /**
+   * Dispatch VISIT_EVENT action on component mount
+   */
+  mounted() {
+    const eventId = this.$route.params.eventId;
+
+    this.$store.dispatch(VISIT_EVENT, { projectId: this.projectId, eventId });
```
Comment (neSpecc) [inline src/components/events/Overview.vue]: maybe we don't need to send request if a user already visited this event

File: src/store/modules/events/index.ts [hunk 4]
```diff
@@ -450,6 +471,23 @@ const module: Module<EventsModuleState, RootState> = {
...
 
+    /**
+     * Mark event as visited in state
+     *
+     * @param {EventsModuleState} state
+     * @param {string} projectId - project event is related to
+     * @param {string} eventId - visited event
+     */
+    [MutationTypes.MARK_AS_VISITED](state, { projectId, eventId }) {
+      const userId = store.state.user.data.id;
```
Comment (neSpecc) [inline src/store/modules/events/index.ts]: what is `store`?

Comment (gohabereg) [reply src/store/modules/events/index.ts]: VueX store

Comment (neSpecc) [reply src/store/modules/events/index.ts]: better to call `this.store` https://github.com/vuejs/vuex/issues/344

Comment (neSpecc) [reply src/store/modules/events/index.ts]: or pass userId from the action. https://github.com/vuejs/vuex/issues/344#issuecomment-249880816

File: src/store/modules/events/index.ts [hunk 5]
```diff
@@ -356,6 +362,23 @@ const module: Module<EventsModuleState, RootState> = {
...
     },
 
+    /**
+     * Send request to mark event as visited
+     *
+     * @param {function} commit
+     * @param {string} projectId - project event is related to
+     * @param {string} eventId - visited event
+     */
+    async [VISIT_EVENT]({ commit }, { projectId, eventId }): Promise<void> {
```
Comment (neSpecc) [inline src/store/modules/events/index.ts]:
```suggestion
    async [VISIT_EVENT]({ commit, rootState }, { projectId, eventId }): Promise<void> {
```

File: src/store/modules/events/index.ts [hunk 6]
```diff
@@ -356,6 +362,23 @@ const module: Module<EventsModuleState, RootState> = {
...
+      const result = await eventsApi.visitEvent(projectId, eventId);
+
+      const userId = (this.state as RootState).user.data.id;
```
Comment (neSpecc) [inline src/store/modules/events/index.ts]:
```suggestion
      const userId = (rootState as RootState).user.data.id;
```

File: src/store/modules/events/index.ts [hunk 7]
```diff
@@ -450,6 +473,23 @@ const module: Module<EventsModuleState, RootState> = {
...
+    /**
+     * Mark event as visited in state
+     *
+     * @param {EventsModuleState} state
+     * @param {string} projectId - project event is related to
+     * @param {string} eventId - visited event
+     * @param {string} userId - user who visited event
+     */
+    [MutationTypes.MARK_AS_VISITED](state, { projectId, eventId, userId }) {
+      const key = projectId + ':' + eventId;
```
Comment (neSpecc) [inline src/store/modules/events/index.ts]: It is not a good practice that forming a key of the `list` object is not in a single place.

**Pull request merged**